hipscat/margin_utils.py

```diff
@@ -290,8 +303,84 @@ def check_margin_bounds(ra, dec, poly_and_wcs):
     sky_coords = SkyCoord(ra, dec, unit='deg')
     bound_vals = []
     for p, w in poly_and_wcs:
-        xv, yv = wcs.utils.skycoord_to_pixel(sky_coords, w)
+        xv, yv = world_coordinate_system.utils.skycoord_to_pixel(sky_coords, w)
         pcs = PixCoord(xv, yv)
         vals = p.contains(pcs)
         bound_vals.append(vals)
-    return np.array(bound_vals).any(axis=0)
\ No newline at end of file
+    return np.array(bound_vals).any(axis=0)
+
+def check_polar_margin_bounds(ra, dec, order, pix, highest_k, pole, margin_threshold, step=1000):
+    part_pix_res = hp.nside2resol(2**order)
+    marg_pix_res = hp.nside2resol(2**highest_k)
+
+    # get the approximate number of boundary samples to cover a highest_k pixel
+    # on the boundary of the main pixel
+    boundary_range = int((marg_pix_res / part_pix_res) * step)
+    pixel_boundaries = hp.vec2dir(hp.boundaries(2**order, pix, step=step, nest=True), lonlat=True)
+
+    if pole == "North":
+        end = len(pixel_boundaries[0])
+        east_ra = pixel_boundaries[0][0:boundary_range+1]
+        east_dec = pixel_boundaries[1][0:boundary_range+1]
+
+        west_ra = pixel_boundaries[0][end-boundary_range:end]
+        west_dec = pixel_boundaries[1][end-boundary_range:end]
+
+        bound_ra = np.concatenate((east_ra, west_ra), axis=None)
+        bound_dec = np.concatenate((east_dec, west_dec), axis=None)
+        polar_bounaries = np.array([bound_ra, bound_dec])
+    else:
+        start = (2 * step) - boundary_range
+        end = (2 * step) + boundary_range + 1
+        south_ra = pixel_boundaries[0][start:end]
+        south_dec = pixel_boundaries[1][start:end]
+        polar_bounaries = np.array([south_ra, south_dec])
+
+    # healpy.boundaries sometimes returns dec values greater than 90, especially
+    # when taking many samples...
+    for i in range(len(polar_bounaries[1])):
```

Comment:
This pattern shows up a couple times, and I think could be replaced with a numpy call to both speed up and make the intent clearer:

```
polar_boundaries[1] = np.clip(polar_boundaries[1], -90., 90.)
```

Reply:
didn't know about this numpy function, thanks!!